### PR TITLE
Token specified for repository

### PR DESCRIPTION
**Pull request by piotr-travisci**

### CODE REVIEW

File: lib/travis/build/data.rb [hunk 1]
```diff
@@ -203,6 +203,11 @@ def repository
         data[:repository] || {}
       end
 
+      def allowed_repositories
+        return data[:allowed_repositories] if data[:allowed_repositories]
```
Comment (BanzaiMan) [inline lib/travis/build/data.rb:207]: This can be more succinctly (and still readably)

    data[:allowed_repositories] || [github_id]

Comment (piotr-travisci) [reply lib/travis/build/data.rb]: Nice, thank you!

**Comment by dentarg** [comment]
@BanzaiMan @piotr-travisci is this related to https://blog.travis-ci.com/clone-or-import-TravisCI?

**Comment by piotrmilcarz** [comment]
@dentarg Yes, it is.

**Comment by dentarg** [comment]
@piotrmilcarz how can we opt-out? since 4 days back, this change breaks our builds, we have one repo cloning many of our other repos, we would like to not have to toggle "Clone or Import" to "ON" for all our repos

It was working with builds that had travis-build version: 1c2611a9

Not working since builds started running with travis-build version: 442f8723

**Comment by piotrmilcarz** [comment]
@dentarg 
You have to turn it ON in all repositories, which are additionally cloned into the build.

**Comment by dentarg** [comment]
@piotrmilcarz is that the only way? why?

**Comment by dentarg** [comment]
@piotrmilcarz FYI we have our own personal access token that have access to the repos

**Comment by piotrmilcarz** [comment]
@dentarg 
For security reasons, we do not use org/user installation token any longer but create the token, which consists of only allowed repositories.

**Comment by dentarg** [comment]
@piotrmilcarz how can we clone our repos in a travis ci build using our own PAT from github?

**Comment by piotrmilcarz** [comment]
@dentarg 
You can also use ssh key for cloning.

**Comment by dentarg** [comment]
@piotrmilcarz I know, but if I want to clone over HTTPS with my own token? isn't that possible on Travis CI?

**Comment by piotrmilcarz** [comment]
@dentarg 
No. We strongly recommend to turn it ON for repositories, which are going to be used in other builds.

**Comment by dentarg** [comment]
@piotrmilcarz why not?

**Comment by mustafa-travisci** [comment]
Hello @dentarg,

You can still use PAT to clone your submodules via HTTPS but what we suggest as my colleague said earlier, using SSH token with the Clone and Import feature is enabled, this is the recommended way to secure your repositories. I hope this helps.

Please reach us out from support@travis-ci.com if you need further assistance for your builds.

Thanks.

**Comment by dentarg** [comment]
@mustafa-travisci @piotrmilcarz how come you consider SSH more secure than HTTPS?

@mustafa-travisci do you have an example on how to do so? (note: it is not submodules in my case, it is other repos)

**Pull request merged**